File: solutions/python/run-length-encoding/1/run_length_encoding.py
def decode(string):

    num = ""
    new = ""

    for ch in string:

        if ch.isdigit():
            num += ch

        else:
            count = int(num) if num else 1
            new += ch * count
            num = ""          

    return new


def encode(string):

    if string == "":
        return ""

    new = ""
    count = 1

    for i in range(1, len(string)):

        if string[i] == string[i - 1]:
            count += 1
        else:
            if count == 1:
                new += string[i - 1]
            else:
                new += str(count) + string[i - 1]

            count = 1

    # last group
    if count == 1:
        new += string[-1]
    else:
        new += str(count) + string[-1]

    return new

File: solutions/javascript/run-length-encoding/1/run-length-encoding.js
//
// This is only a SKELETON file for the 'Run Length Encoding' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const encode = (str) => {
  let result='';
  let count=1;


  for (let ch in str) {
    if (str[ch] === str[parseInt(ch) + 1]) {
      count = count + 1;
    } else {
      if (count > 1) {
        result = result + count + str[ch];
      } else {
        result = result + str[ch];
      }
      count = 1; 
    }
  }

  return result;
};

export const decode = function(str) {
  let result = '';
  let num = '';

  for (let ch of str) {
    if (ch >= '0' && ch <= '9') {
      num += ch;  
    } else {
    
      let count = num ? parseInt(num, 10) : 1;
      result += ch.repeat(count);
      num = '';
    }
  }

  return result;
}


